app/scripts/modules/core/pipeline/config/triggers/docker/dockerTriggerOptions.directive.js

```diff
@@ -0,0 +1,68 @@
+'use strict';
+
+const angular = require('angular');
+
+module.exports = angular
+  .module('spinnaker.core.pipeline.config.triggers.docker.options.directive', [
+    require('../../../../../core/config/settings.js'),
+  ])
+  .directive('dockerTriggerOptions', function () {
+    return {
+      restrict: 'E',
+      templateUrl: require('./dockerTriggerOptions.directive.html'),
+      bindToController: {
+        command: '=',
+      },
+      controller: 'dockerTriggerOptionsCtrl',
+      controllerAs: 'vm',
+      scope: {}
+    };
+  })
+  .controller('dockerTriggerOptionsCtrl', function ($scope, Restangular) {
+    // These fields will be added to the trigger when the form is submitted
+    this.command.extraFields = {};
+
+    this.viewState = {
+      tagsLoading: false,
+      loadError: false,
+      selectedtag: null,
+    };
+
+    let tagLoadSuccess = (tags) => {
+      this.tags = tags.map((val) => val.tag ).sort();
+      if (this.tags.length) {
+        let defaultSelection = this.tags[0];
+        this.viewState.selectedtag = defaultSelection;
+        this.updateSelectedtag(defaultSelection);
+      }
+      this.viewState.tagsLoading = false;
+    };
+
+    let tagLoadFailure = () => {
+      this.viewState.tagsLoading = false;
+      this.viewState.loadError = true;
+    };
+
+    let initialize = () => {
+      let command = this.command;
+      // do not re-initialize if the trigger has changed to some other type
+      if (command.trigger.type !== 'docker') {
+        return;
+      }
+      this.viewState.tagsLoading = true;
+
+      Restangular.all('images/find').getList({ provider: 'dockerRegistry', account: command.trigger.account, q: command.trigger.repository }, {}).then(function(results) {
```

Comment:
I don't understand what you are trying to avoid modifying here - is it the `loadImages` method in `DockerTriggerCtrl`?

It would be much better to get this out of the controller and into a service - same with `DockerTriggerCtrl#loadImages`